server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -123,10 +123,14 @@ private RestRequest(
         this.requestId = requestId;
     }
 
-    protected RestRequest(RestRequest other) {
+    // public for test
```

Comment:
I think it would be preferable to use an anonymous class in `RestRequestFilterTests` and keep this constructor `protected`

Reply:
Nice hint, thanks!
This trick for circumventing protected member access on public classes is not something I think about, I admit. Looks preferable indeed, I'll keep it in mind further.

Reply:
3d23eff